src/lib/signature_kind/mainnet/mina_signature_kind.ml

```diff
@@ -1,3 +1,3 @@
 include Mina_signature_kind_type
 
-let t_DEPRECATED = Mainnet
+let t_DEPRECATED = Mainnet [@@deriving bin_io_unversioned]
```

Comment:
```suggestion
let t_DEPRECATED = Mainnet
```

It doesn't make sense to add this deriving to a value; it should be added to the type. I'm not sure how it was working for you in terms of compilation.